pkg/cmd/repo/autolink/http.go

```diff
@@ -0,0 +1,44 @@
+package autolink
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+)
+
+func repoAutolinks(httpClient *http.Client, repo ghrepo.Interface) ([]autolink, error) {
+	path := fmt.Sprintf("repos/%s/%s/autolinks", repo.RepoOwner(), repo.RepoName())
+	url := ghinstance.RESTPrefix(repo.RepoHost()) + path
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode > 299 {
+		return nil, api.HandleHTTPError(resp)
+	}
```

Comment:
The AC mentions helpful error messaging for 404's

```
error getting autolinks: HTTP 404: Must have admin rights to Repository. (https://api.github.com/repos/{owner}/{repo}/autolinks)
```

I think that would require an extra conditional, here:

```suggestion
        if resp.StatusCode == 404 {
		return nil, fmt.Errorf("error getting autolinks: HTTP 404: Must have admin rights to Repository. (https://api.github.com/repos/%s)", path)
	} else if resp.StatusCode > 299 {
		return nil, api.HandleHTTPError(resp)
	}
```

Note, this suppresses the actual error returned by the api for 404's. Maybe that's intentional, but I'd look to @andyfeller to chime in, here.

No need to block merging this on his response, though. We can always open a follow-up PR to change the error handling if the AC needs revisiting 🙂

Reply:
This messaging was added via https://github.com/hoffm/github-cli/pull/1. I'll leave the comment unresolved for @andyfeller's benefit.

Reply:
Interesting 🤔 when I was originally authoring the AC, I was basing this on the response provided back by `gh api` making the necessary calls.

@jtmcg makes a good point that it needs to be easy for users to understand if they don't have the right permission and what they should do to correct it.  In this case, it isn't something trivial like `gh auth refresh -s <SCOPE>` but the role assigned to you.

Let me catch up on the rest of these changes as I trust y'all worked something up while I was OoO.